pkg/ocm/ocm.go

```diff
@@ -336,3 +292,68 @@ func (c *Client) TransferComponent(ctx context.Context, obj *v1alpha1.ComponentS
 
 	return nil
 }
+
+// maybeConfigureAccessCredentials configures access credentials if needed for a source/destination repository.
+func (c *Client) maybeConfigureAccessCredentials(ctx context.Context, ocmCtx ocm.Context, repository v1alpha1.OCMRepository, namespace string) error {
+	// If there are no credentials, this call is a no-op.
+	if repository.SecretRef == nil {
+		return nil
+	}
+
+	logger := log.FromContext(ctx)
+
+	if err := csdk.ConfigureCredentials(ctx, ocmCtx, c.client, repository.URL, repository.SecretRef.Name, namespace); err != nil {
+		logger.V(4).Error(err, "failed to find destination credentials")
+
+		// we don't ignore not found errors
+		return fmt.Errorf("failed to configure credentials for component: %w", err)
+	}
+
+	logger.V(4).Info("credentials configured")
+
+	return nil
+}
+
+func (c *Client) maybeConfigureServiceAccountAccess(ctx context.Context, octx ocm.Context, obj *v1alpha1.ComponentSubscription) error {
+	logger := log.FromContext(ctx)
+
+	logger.V(4).Info("configuring service account credentials")
+
+	if obj.Spec.ServiceAccountName == "" {
+		return nil
+	}
+
```

Comment:
could this check be outside the method? then you only need to pass in the service account name and the contexts.

Reply:
I'm also using `obj.Namespace,`. But I can pass that in instead. Would minimize the memory burden.